inference-engine/tests/unit/vpu/middleend_tests/passes_tests/adjust_data_batch_tests.cpp

```diff
@@ -0,0 +1,389 @@
+            successors.push_back(output);
+
+            if (expected == StageType::LoopStart) {
+                EXPECT_EQ(consumer->numOutputs(), 1);
+                EXPECT_EQ(output->desc().dim(Dim::N), 1);
+            } else if (expected == StageType::LoopEnd) {
+                EXPECT_EQ(output->desc().dim(Dim::N), batchSize);
+            }
+        }
+
+        return successors;
+    }
+
+    DataVector checkSingleLoopEnd(const Data& data) {
+        EXPECT_EQ(data->numConsumers(), 1);
+
+        const auto& consumer = data->singleConsumer();
+        EXPECT_EQ(consumer->type(), StageType::LoopEnd);
+        DataVector outputs;
+        for (const auto& output : consumer->outputs()) {
+            EXPECT_EQ(output->desc().dim(Dim::N), batchSize);
+            outputs.push_back(output);
+        }
+
+        return outputs;
+    }
+
+    static Data CheckSingleConnection(const Data& data, int testInd, int batch = 1) {
+        EXPECT_EQ(data->numConsumers(), 1);
+
+        const auto& consumer = data->singleConsumer();
+        EXPECT_EQ(consumer->type(), StageType::None);
+        EXPECT_EQ(consumer->attrs().get<int>("test_ind"), testInd);
+        EXPECT_EQ(consumer->numOutputs(), 1);
+        const auto& output = consumer->output(0);
+        EXPECT_EQ(output->desc().dim(Dim::N), batch);
+        return output;
+    }
+
+    static Data singleElement(const DataVector& dataObjects) {
+        EXPECT_EQ(dataObjects.size(), 1);
+        return dataObjects.front();
+    }
+};
+
+TEST_F(VPU_AdjustDataBatchTest, Case1_LinearWithBatchedInTheEnd) {
+    //
+    // [Input] -> (Split) -> (Split) -> (Split) -> (Split) -> (Split) -> (Split) -> (Batched) -> [Output]
+    //
+
+    testModel.createInputs(1);
+    testModel.createOutputs(1);
+
+    testModel.addStage({InputInfo::fromNetwork()}, {OutputInfo::intermediate()}, true);
+    testModel.setStageBatchInfo(0, {{0, BatchSupport::Split}});
+
+    testModel.addStage({InputInfo::fromPrevStage(0)}, {OutputInfo::intermediate()}, true);
+    testModel.setStageBatchInfo(1, {{0, BatchSupport::Split}});
+
+    testModel.addStage({InputInfo::fromPrevStage(1)}, {OutputInfo::intermediate()}, true);
+    testModel.setStageBatchInfo(2, {{0, BatchSupport::Split}});
+
+    testModel.addStage({InputInfo::fromPrevStage(2)}, {OutputInfo::intermediate()}, true);
+    testModel.setStageBatchInfo(3, {{0, BatchSupport::Split}});
+
+    testModel.addStage({InputInfo::fromPrevStage(3)}, {OutputInfo::intermediate()}, true);
+    testModel.setStageBatchInfo(4, {{0, BatchSupport::Split}});
+
+    testModel.addStage({InputInfo::fromPrevStage(4)}, {OutputInfo::intermediate()}, true);
+    testModel.setStageBatchInfo(5, {{0, BatchSupport::Split}});
+
+    testModel.addStage({InputInfo::fromPrevStage(5)}, {OutputInfo::fromNetwork()}, true);
+
+    RunPass();
+
+    const auto& data0 = singleElement(checkSingleLoopStart(testModel.getInputs().at(0)));
+    const auto& data1 = CheckSingleConnection(data0, 0);
```

Comment:
The same there. Use `for` loop.

Reply:
Why this discussion was resolved?

Reply:
After some thoughts I concluded that in this particular case code reduction may harm the readability of the test, so I didn't insist on this change and resolved this.